plugin-api/src/main/java/org/hyperledger/besu/plugin/services/storage/PrivacyKeyValueStorageFactory.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.plugin.services.storage;
+
+public interface PrivacyKeyValueStorageFactory extends KeyValueStorageFactory {
+  /**
+   * Retrieves the schema version of the key-value storage factory.
+   *
+   * @return the schema version of the key-value storage factory.
+   */
+  int getDatabaseVersion();
```

Comment:
nit; rename to `getSchemaVersion`, this would be to avoid any potential confusion, as with relational dbs often schema != database version

Reply:
Interesting! I actually want to encapsulate both schema and database version in a single parameter. E.g. v0 -> v1 is a schema version change but v1 -> v2 will most likely be a db version change (segmenting the PrivateStateStorage). I will think about this some more to make it clear.

Reply:
Are you wanting to support older versions of the database with newer versions of the schema ...or force database upgrades to use the latest schema?
_(that decision will likely impact your choice on this matter 🤔)_

Reply:
That logic can be implemented in Besu. In v0 -> v1 there is no choice.

Reply:
I made it `@Unstable`